src/Libraries/Microsoft.Extensions.Http.Resilience/Resilience/Internal/Validators/HttpStandardResilienceOptionsValidator.cs

```diff
@@ -2,11 +2,10 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using Microsoft.Extensions.Options;
-using Microsoft.Extensions.Options.Validation;
 
 namespace Microsoft.Extensions.Http.Resilience.Internal.Validators;
 
-[OptionsValidator]
-internal sealed partial class HttpStandardResilienceOptionsValidator : IValidateOptions<HttpStandardResilienceOptions>
+internal sealed class HttpStandardResilienceOptionsValidator : IValidateOptions<HttpStandardResilienceOptions>
```

Comment:
@geeknoid I had to disable this validator because it materializes even the internal validation attributes we have in Polly and the PR won't compile. 

I believe the generator should just skip those.

Reply:
Does the option type in this package derive from the option type in the Polly package?

@tarekgh This looks like a case we hadn't seen before and since you know own the option validation generator, well, here you go :-) I don't know if there's a way to compose this together such that the derived option type gets full validation even for the properties inherited from the base class.

Reply:
> Does the option type in this package derive from the option type in the Polly package?

Yes, this is the internal attribute we are using:

https://github.com/App-vNext/Polly/blob/a9bf3e95a37ff9c98ca02c8ca395f91645ffe5d8/src/Polly.Core/Timeout/TimeoutStrategyOptions.cs#L20

Reply:
I don't think this is going to work without you publishing something more in the polly code. Otherwise, there is no way for the validation code generated in this assembly to validate the state in the base class.

If you make TimeoutAttribute public, then this will just work as the generated code will have access to the atribute definition.

In order to keep the attribute hidden, then you'd need to make public instead the generated options validation type from the Polly library so that it could be called from this assembly.

Perhaps there would be a way for Polly to register an option validator for its part of the state, and then this assembly only validates its part of the state?

In any case, @tarekgh, we should fix the option generator to ignore attributes that aren't reachable from the generated code.

Reply:
Ideally, it would be nice to get rid of it and use built-in attributes.

I need to validate that timespan is in range of -1 milliseconds to max. Also, zero or default value must be excluded.

Reply:
Could you provide isolated reproduce for this problem so I can look at it?

Reply:
Here a repro: https://github.com/dotnet/extensions/commit/1bdd9251a72090b321cd742c08466de127d8413d

@martintmk I think your best bet for now is to make the attribute public in Polly and then it'll work correctly.

Reply:
@geeknoid Thanks Martin for the repro!

I have actually took a second look at our custom attributes and realized we can actually drop them:
https://github.com/App-vNext/Polly/pull/1351

Still, the generator behavior should properly handle internal attributes.